edgedb/protocol/codecs/codecs.pyx

```diff
@@ -451,6 +451,35 @@ cdef relative_duration_decode(pgproto.CodecContext settings, FRBuffer *buf):
         microseconds=microseconds, days=days, months=months)
 
 
+cdef date_duration_encode(pgproto.CodecContext settings, WriteBuffer buf,
+                          object obj):
+
+    cdef:
+        days = obj.days
+        months = obj.months
+
+    buf.write_int32(16)
+    buf.write_int64(0)
+    buf.write_int32(days)
+    buf.write_int32(months)
+
+
+cdef date_duration_decode(pgproto.CodecContext settings, FRBuffer *buf):
+    cdef:
+        int32_t days
+        int32_t months
+        int64_t microseconds
```

Comment:
Why are there microseconds here? Shouldn't the server reliably send the correct data type, the one without fractional day parts?

Reply:
See the assertion below.

Reply:
To clarify, there is no "data type without fractions" on the wire, we use `interval` for all durations.